docs/06.REFERENCE-COUNTING.md

```diff
@@ -0,0 +1,156 @@
+## Reference counting in JerryScript
+
+In JerryScript all values (jerry_value_t) are independent
```

Comment:
I'd use ` (code insertion) to highlight types and function names instead of * (*italic*)